src/service/data_products/samples.py

```diff
@@ -159,6 +166,27 @@ class Samples(BaseModel):
     )
 
 
+@_ROUTER.get(
+    "/meta",
+    response_model=col_models.ColumnarAttributesMeta,
+    description=
+"""
+Get metadata about the samples table including column names, type,
+minimum and maximum values, etc.
+""")
+async def get_samples_meta(
```

Comment:
Hmm, it's kind of pointless to have the non-visible key in the results because it's always going to be false and is useless for the user.

maybe make a another pydantic class that inherits from the spec class and just adds that field, use that to load the specs, and then covert to the returned class which will no longer have that field.

Reply:
let's do it in the next PRs? I think that will be a quite large change. And I haven't completely think it through. 
say we have `AttributesColumnSpecLoading` and `AttributesColumnSpec` with non_visible field that inherits `AttributesColumnSpecLoading`. Then we make `AttributesColumn` and `AttributesColumnLoading`? And `ColumnarAttributesMeta` and `ColumnarAttributesMetaLoading`?

Reply:
I don't think it's that big a deal unless I haven't thought this through correctly. I'd do it w/ 3 classes:
* AttribsColumnSpecBase - no non-visible field, otherwise the same as current AttribsColumnSpec
* AttribsColumnSpec - inherits from Base, adds non-visible field & nothing else
* AttributesColumn - inherits from Base, otherwise no changes

I don't think the other models would need changes in that case.

Reply:
AttributesColumn needs to non_visible field. Because the `/meta` endpoint is only returning non-visible columns.

Reply:
```
    doc[col_models.FIELD_COLUMNS] = [col_models.AttributesColumn(**d)
                                     for d in doc[col_models.FIELD_COLUMNS]]
```
to
```
    doc[col_models.FIELD_COLUMNS] = [col_models.AttributesColumn(**d)
                                     for d in doc[col_models.FIELD_COLUMNS]
                                     if not d[col_models.NON_VISIBLE]
    ]
```
or something like that should work I think

Reply:
👍 I see. Yea. That works.

Reply:
:+1: